dracut/30ignition/coreos-gpt-setup.service

```diff
@@ -1,14 +1,20 @@
 [Unit]
-Description=Generate new UUID for disk GPT %I
+Description=Generate new UUID for boot disk GPT
 Documentation=https://github.com/coreos/ignition
 ConditionPathExists=/etc/initrd-release
 DefaultDependencies=no
 Before=local-fs-pre.target systemd-fsck-root.service
 Before=ignition-diskful.target
 Wants=systemd-udevd.service
 After=systemd-udevd.service
-Requires=%i.device
-After=%i.device
+
+# Make sure that we wait for the disk with the partition labeled 'boot' to come
+# up. This is the only unit where it is safe to wait on a specific disk label.
+# Units that requires specific partitions to be available should order
+# themselves after ignition-diskful.target.
```

Comment:
Did you mean `ignition-disks.service`?

Reply:
As far as I understand, you want to order after `ignition-diskful.target` to work on 1rst boot available partitions and after `ignition-disks.service` for ignition provisioned partitions. Maybe I should clarify the comment.

Reply:
I'd say it like anything that is operating on "persistent bootable disk" needs `-diskful.target`, as does this.  We probably don't need to describe that in a comment here, the idea should be covered in the unit docs.

What I would say in this comment is two things:

- First, we don't have any plans to support re-provisioning/re-writing the `/boot` partition.  So the disk holding it must be the root.
- We can't use `root` because RHCOS's pre-setup LUKS container doesn't allow it

This all said...I am increasingly feeling we should make this logic part of `ignition-disks.service` along with https://github.com/coreos/fedora-coreos-config/pull/354 or so.